11-rmarkdown-application-regression.Rmd

```diff
@@ -0,0 +1,217 @@
+---
+title: "Further Application of R Markdown"
+teaching: 45
+exercises: 15
+questions:
+- "How can I use R Markdown to integrate analysis results in my report?"
+objectives:
+- "Introduction to regression with lm()"
+- "Demonstrate how to add a regression line and statistics to a ggplot"
+- "Introduction to the `broom` package for 'tidying' statistical results"
+- "Demonstrate how to include statistical results in a reproducible document"
+source: Rmd
+---
+
+```{r, include=FALSE}
+source("bin/chunk-options.R")
+knitr_fig_path("11-")
+```
+
+# Applications of R Markdown
+
+Now that we've learned the basics of R and R Markdown, we're going to tie some 
+things together by doing a regression on the Gapminder data and displaying our
+results in a nicely formatted document.
+
+Working in a new R Markdown document (File -> New File -> R Markdown Document), 
+we can load the libraries that we are going to use, and load the data in 
+a code chunk:
+
+<pre>
+&#96;&#96;&#96;{r load}
+library(readr)
+library(dplyr)
+library(ggplot2)
+gapminder <- read_csv("data/gapminder_data.csv")
+&#96;&#96;&#96;
+</pre>
+
+For now, we are just going to use the data for Canada, so let's insert a code
+chunk to filter our data to just Canada:
+
+<pre>
+&#96;&#96;&#96;{r filter-can}
+canada <- filter(gapminder, country == "Canada")
+&#96;&#96;&#96;
+</pre>
+
+We want to do a regression investigating the relationship between life
+expectancy and per-capita GDP in Canada. 
+
+We can start by inserting another code chunk to do a quick visualization of the
+data. We can insert the regression line on the points, but this doesn't give us
+the regression coefficients, significance, etc. that we need to really evaluate
+the relationship.
+
+<pre>
+&#96;&#96;&#96;{r plot-1}
+ggplot(canada, aes(x = lifeExp, y = gdpPercap)) + 
+  geom_point() + 
+  geom_smooth(method = "lm")
+&#96;&#96;&#96;
+</pre>
+
+We use the `lm()` function to perform a General Linear Model (GLM), of which 
+a regression is one form. It is specified by a new syntax, a "formula" syntax:
+
+```r
+lm(response ~ predictor(s), data = data, ...)
+```
+
+So in our case, we can put in a new code chunk to fit the model: 
+
+<pre>
+&#96;&#96;&#96;{r model}
+canada_lm <- lm(gdpPercap ~ lifeExp, data = canada)
+summary(canada_lm)
+&#96;&#96;&#96;
+</pre>
+
+The `summary()` function gives us the high level information from our regression 
+model: The coefficient estimates, the R-squared, and the p-value.
+
+And the `plot()` function gives us some basic diagnostic plots:
+
+<pre>
+&#96;&#96;&#96;{r diagnostics}
+plot(canada_lm)
+&#96;&#96;&#96;
+</pre>
+
+But these `lm` objects are ungainly - they are fine to look at to inspect the 
+important parts of our model, but difficult to extract those pieces that we 
+want to use later - such as regression coefficients and P values.
+
+There is another package we can use, called *broom*, that is used to tidy
+model results into a format we are familiar with: A `data.frame`:
```

Comment:
```suggestion
model results into a format we are familiar with: a `data.frame`. 
```
Also is it worth mentioning that broom works on many types of models?

Reply:
👍